### PR TITLE
Updated gazebo PID format

### PR DESCRIPTION
**Pull request by Yeshasvitvs**
See https://github.com/robotology/icub-gazebo/issues/44

### CODE REVIEW

File: icub/conf/gazebo_icub_left_arm_no_hand.ini [hunk 1]
```diff
@@ -29,28 +29,33 @@ name left_arm_no_hand
...
+
+[VELOCITY_CONTROL]
+controlUnits  metric_units
+controlLaw    joint_pid_gazebo_v1
+kp 0. 8.7266388888888873 8.7266388888888873 8.7266388888888873 5.2359833333333325 5.2359833333333325 5.2359833333333325 5.2359833333333325
+kd 0.034906555555555552 0.034906555555555552 0.034906555555555552 0.0017453277777777776 0.0017453277777777776 0.0017453277777777776 0.0017453277777777776
+ki 0.0017453277777777776 0.0017453277777777776 0.0017453277777777776 0.00017453277777777776 0.00017453277777777776 0.00017453277777777776 0.00017453277777777776
+maxInt 9999 9999 9999 9999 9999 9999 9999
+maxOutput 9999 9999 9999 9999 9999 9999 9999
+shift
```
Comment (traversaro) [inline icub/conf/gazebo_icub_left_arm_no_hand.ini]: I think this should be 0 zeros

**Comment by traversaro** [comment]
I don't know if it was intentional, but if you could revert the changes on the whitespaces such that the column remain aligned (for example in https://github.com/robotology/icub-gazebo/pull/51/files#diff-6091a83cb3686504569fbc359129add2L43) it would be great for the maintainability .

For the same reason (maintainability) it would be convenient to reduce the number of decimal numbers in the gains, after a first inspection I think 3 decimal number should be enough for most uses.

**Comment by traversaro** [comment]
Note: this commit fixes also a problem related to https://github.com/robotology/icub-gazebo/issues/8 (camera were not respecting the naming policy).

File: icub/conf/gazebo_icub_head.ini [hunk 2]
```diff
@@ -23,34 +23,39 @@ device gazebo_controlboard
...
-Pid5 250.0 0.0 0.1 9999 9999 9 9
+[POSITION_CONTROL]
+controlUnits  metric_units
+controlLaw    joint_pid_gazebo_v1
+kp 1.7453277777777776 1.7453277777777776 1.7453277777777776 4.3633194444444436 4.3633194444444436 4.3633194444444436
+kd 0.12217294444444443 0.12217294444444443 0.12217294444444443 0. 0. 0.
+ki 0.0034906555555555552 0.0034906555555555552 0.0034906555555555552 0.0017453277777777776 0.0017453277777777776 0.0017453277777777776
+maxInt 9999 9999 9999 9999 9999 9999
+maxOutput 9999 9999 9999 9999 9999 9999
+shift
```
Comment (randaz81) [inline icub/conf/gazebo_icub_head.ini]: Those parameters require the same number of columns of the others 
e.g. in this specific example six columns are required:  0.0 0.0 0.0 0.0 0.0 0.0

File: icub/conf/gazebo_icub_head.ini [hunk 3]
```diff
@@ -23,34 +23,39 @@ device gazebo_controlboard
...
+
+[VELOCITY_CONTROL]
+controlUnits  metric_units
+controlLaw    joint_pid_gazebo_v1
+kp 8.7266388888888873 8.7266388888888873 8.7266388888888873 4.3633194444444436 4.3633194444444436 4.3633194444444436
+kd 0.034906555555555552 0.034906555555555552 0.034906555555555552 0. 0. 0.
+ki 0.0034906555555555552 0.0034906555555555552 0.0034906555555555552 0.0017453277777777776 0.0017453277777777776 0.0017453277777777776
+maxInt 9999 9999 9999 9999 9999 9999
+maxOutput 9999 9999 9999 9999 9999 9999
+shift
```
Comment (randaz81) [inline icub/conf/gazebo_icub_head.ini]: also here... etc..

File: icub/conf/gazebo_icub_left_hand_finger.ini [hunk 4]
```diff
@@ -19,28 +19,41 @@ fingers_abduction_control (0 1 2 3) (l_hand_fingers reserved reserved reserved r
...
+kp 1.7453277777777776 1.7453277777777776 1.7453277777777776 1.7453277777777776
+kd 0.00017453277777777776 0.00017453277777777776 0.00017453277777777776 0.00017453277777777776
+ki 0.17453277777777776 0.17453277777777776 0.17453277777777776 0.17453277777777776
+maxInt 9999 9999 9999 9999
+maxOutput 9999 9999 9999 9999
+shift        
+ko           
+stictionUp   
+stictionDwn  
+Pid3 100.0 0.01 10.0 9999 9999 9 9
```
Comment (randaz81) [inline icub/conf/gazebo_icub_left_hand_finger.ini]: stray line Pid3 should be removed

File: icub/conf/gazebo_icub_left_hand_index.ini [hunk 5]
```diff
@@ -19,24 +19,40 @@ index_control (0 1 2) (l_index_proximal l_index_distal reserved)
...
+kp 1.7453277777777776 1.7453277777777776 1.7453277777777776
+kd 0.00017453277777777776 0.00017453277777777776 0.00017453277777777776
+ki 0.17453277777777776 0.17453277777777776 0.17453277777777776
+maxInt 9999 9999 9999
+maxOutput 9999 9999 9999
+shift        
+ko           
+stictionUp   
+stictionDwn  
+Pid2 100.0 0.01 10.0 9999 9999 9 9
```
Comment (randaz81) [inline icub/conf/gazebo_icub_left_hand_index.ini]: Also here (Pid2). Same problem in few other files below

Comment (Yeshasvitvs) [inline icub/conf/gazebo_icub_left_hand_index.ini]: oopse! scripting mistakes, will fix them @randaz81

**Comment by Yeshasvitvs** [comment]
@traversaro  @randaz81  please review the latest changes

File: icub/conf/gazebo_icub_right_hand_pinky.ini [hunk 6]
```diff
@@ -19,29 +19,38 @@ pinky_control (0 1 2 3 4 5) (r_pinky reserved reserved reserved reserved reserve
...
-Pid1  500.0 2.0 0.1  9999 9999 9 9
-Pid2  500.0 2.0 0.1  9999 9999 9 9
-Pid3  500.0 2.0 0.1  9999 9999 9 9
-Pid4  500.0 2.0 0.1  9999 9999 9 9
-Pid5  500.0 2.0 0.1  9999 9999 9 9
+[POSITION_CONTROL]
+controlUnits  metric_units
+controlLaw    joint_pid_gazebo_v1
+kp            1.745 1.745 1.745 1.745 1.745 1.745
+kd            0.0   0.0   0.0   0.0   0.0   0.0
```
Comment (traversaro) [inline icub/conf/gazebo_icub_right_hand_pinky.ini]: @Yeshasvitvs Have you check that setting this `kd` to zero (they where different from zero before) is not affecting the behaviour of the simulation?

File: icub/conf/gazebo_icub_right_leg.ini [hunk 7]
```diff
@@ -37,10 +37,10 @@ kd            0.174  0.174  0.174  0.174  0.174  0.174
...
 maxInt        9999   9999   9999   9999   9999   9999
 maxOutput     9999   9999   9999   9999   9999   9999
-shift         0.0   0.0   0.0   0.0   0.0   0.0   0.0
-ko            0.0   0.0   0.0   0.0   0.0   0.0   0.0
-stictionUp    0.0   0.0   0.0   0.0   0.0   0.0   0.0
-stictionDwn   0.0   0.0   0.0   0.0   0.0   0.0   0.0
+shift         0.0    0.0    0.0    0.0    0.0    0.0
+ko            0.0    0.0    0.0    0.0    0.0    0.0
+stictionUp    0.0    0.0    0.0    0.0    0.0    0.0
+stictionDwn   0.0    0.0    0.0    0.0    0.0    0.0
```
Comment (traversaro) [inline icub/conf/gazebo_icub_right_leg.ini]: 👍

File: icub/conf/gazebo_icub_right_hand_pinky.ini (hunk 6 above)
Comment (Yeshasvitvs) [inline icub/conf/gazebo_icub_right_hand_pinky.ini]: @traversaro I checked the motion of all the joint on both hands with the currently set PID values and it works fine. If there is nothing else we can merge this. 

This should also close https://github.com/robotology/icub-gazebo/issues/24

**Pull request merged**